### PR TITLE
Swift3

### PR DESCRIPTION
**Pull request by jollyjoester**
Please review 😃

### CODE REVIEW

**Comment by giginet** [comment]
Thank you for your contribution 🙏 

I'll review this PR in this weekend
Please moment.

**Comment by jollyjoester** [comment]
Oops... I forgot removing Xcode7.3.1 environment in TravisCI.
I'll do it later.

File: CustomKeyboardTextFieldDemo/CustomKeyboard.swift [hunk 1]
```diff
@@ -24,25 +28,24 @@ class PokemonPickerKeyboard: UIPickerViewKeyboard<PokemonPickerKeyboardDataSourc
         return view
     }
 
-    func pickerView(pickerView: UIPickerView, rowHeightForComponent component: Int) -> CGFloat {
+    func pickerView(_ pickerView: UIPickerView, rowHeightForComponent component: Int) -> CGFloat {
         return rowHeight
     }
 
-    private func iconView(row: Int) -> UIImageView {
+    fileprivate func iconView(_ row: Int) -> UIImageView {
```
Comment (giginet) [inline CustomKeyboardTextFieldDemo/CustomKeyboard.swift]: `private`

File: CustomKeyboardTextFieldDemo/CustomKeyboard.swift [hunk 2]
```diff
@@ -24,25 +28,24 @@ class PokemonPickerKeyboard: UIPickerViewKeyboard<PokemonPickerKeyboardDataSourc
...
         let pokemonName = pickerView(self, titleForRow: row, forComponent: 0)!
         let image = UIImage(named: "\(pokemonName).png")
         return UIImageView(image: image)
     }
 
-    private func updateTextField(for row: Int) {
+    fileprivate func updateTextField(for row: Int) {
```
Comment (giginet) [inline CustomKeyboardTextFieldDemo/CustomKeyboard.swift]: `private`

File: CustomKeyboardTextFieldDemo/CustomKeyboard.swift [hunk 3]
```diff
@@ -105,7 +108,7 @@ class GamePadKeyboardView: UIView, CustomKeyboardView {
         }
     }
 
-    private func insertText(insertText: String) {
+    fileprivate func insertText(_ insertText: String) {
```
Comment (giginet) [inline CustomKeyboardTextFieldDemo/CustomKeyboard.swift]: `private`

File: Sources/CustomKeyboardTextField.swift [hunk 4]
```diff
@@ -42,7 +42,11 @@ public class CustomKeyboardTextField<KeyboardViewProvider: CustomKeyboardProvide
         setupCustomKeyboard()
     }
 
-    private func setupCustomKeyboard() {
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    fileprivate func setupCustomKeyboard() {
```
Comment (giginet) [inline Sources/CustomKeyboardTextField.swift]: `private`

File: Sources/CustomKeyboardTextField.swift [hunk 5]
```diff
@@ -13,18 +13,18 @@ extension UITextField: CustomKeyboardTextFieldType {
...
 }
 
-public class CustomKeyboardTextField<KeyboardViewProvider: CustomKeyboardProvider>: UITextField, UITextFieldDelegate {
+open class CustomKeyboardTextField<KeyboardViewProvider: CustomKeyboardProvider>: UITextField, UITextFieldDelegate {
     var provider: CustomKeyboardProvider {
         return KeyboardViewProvider()
     }
 
-    private(set) var customKeyboardView: CustomKeyboardView? {
+    fileprivate(set) var customKeyboardView: CustomKeyboardView? {
```
Comment (giginet) [inline Sources/CustomKeyboardTextField.swift]: private

File: Sources/CustomKeyboardTextField.swift [hunk 6]
```diff
@@ -13,18 +13,18 @@ extension UITextField: CustomKeyboardTextFieldType {
...
         didSet {
             inputView = customKeyboardView as? UIView
         }
     }
 
-    private(set) var customKeyboardAccessoryView: CustomKeyboardAccessoryView? {
+    fileprivate(set) var customKeyboardAccessoryView: CustomKeyboardAccessoryView? {
```
Comment (giginet) [inline Sources/CustomKeyboardTextField.swift]: private

File: Sources/Keyboards/ToolbarAccessoryView.swift [hunk 7]
```diff
@@ -1,9 +1,9 @@
 import UIKit
 
-public class ToolbarKeyboardAccessoryView: UIToolbar, CustomKeyboardAccessoryView {
+open class ToolbarKeyboardAccessoryView: UIToolbar, CustomKeyboardAccessoryView {
     var toolbar = UIToolbar()
     let toolbarHeight: CGFloat = 44.0
-    weak public var textField: UITextField? = nil
+    weak open var textField: UITextField? = nil
```
Comment (giginet) [inline Sources/Keyboards/ToolbarAccessoryView.swift]: `public`

File: Sources/Keyboards/UIDatePickerKeyboard.swift [hunk 8]
```diff
@@ -2,63 +2,63 @@ import UIKit
...
     var timeFormat: String {
         return "yyyy/MM/dd HH:mm:ss Z"
     }
 }
 
-public class UIDatePickerKeyboard<DataSource: UIDatePickerKeyboardDataSource>: UIDatePicker, CustomKeyboardView {
-    public let datePickerKeyboardDataSource: DataSource
-    weak public var textField: UITextField? = nil
+open class UIDatePickerKeyboard<DataSource: UIDatePickerKeyboardDataSource>: UIDatePicker, CustomKeyboardView {
+    open let datePickerKeyboardDataSource: DataSource
```
Comment (giginet) [inline Sources/Keyboards/UIDatePickerKeyboard.swift]: public

File: Sources/Keyboards/UIDatePickerKeyboard.swift [hunk 9]
```diff
@@ -2,63 +2,63 @@ import UIKit
...
+    weak open var textField: UITextField? = nil
```
Comment (giginet) [inline Sources/Keyboards/UIDatePickerKeyboard.swift]: public

File: Sources/Keyboards/UIPickerViewKeyboard.swift [hunk 10]
```diff
@@ -12,9 +12,9 @@ public extension UIPickerViewKeyboardDataSource {
     }
 }
 
-public class UIPickerViewKeyboard<DataSource: UIPickerViewKeyboardDataSource>: UIPickerView, UIPickerViewDelegate, CustomKeyboardView, UIPickerViewDataSource {
-    public let pickerKeyboardDataSource: DataSource
-    weak public var textField: UITextField? = nil
+open class UIPickerViewKeyboard<DataSource: UIPickerViewKeyboardDataSource>: UIPickerView, UIPickerViewDelegate, CustomKeyboardView, UIPickerViewDataSource {
+    open let pickerKeyboardDataSource: DataSource
```
Comment (giginet) [inline Sources/Keyboards/UIPickerViewKeyboard.swift]: public

File: Sources/Keyboards/UIPickerViewKeyboard.swift [hunk 11]
```diff
@@ -12,9 +12,9 @@ public extension UIPickerViewKeyboardDataSource {
...
+    weak open var textField: UITextField? = nil
```
Comment (giginet) [inline Sources/Keyboards/UIPickerViewKeyboard.swift]: public

**Review by giginet — CHANGES_REQUESTED** [review]
Some access control attributes should be closer.

And please fix CI.
You should drop Xcode 7.x support then It might be fixed

**Comment by jollyjoester** [comment]
I fixed my code that you pointed out.
And I'm not familiar with TravisCI, so please check .travis.yml.

**Comment by giginet** [comment]
@jollyjoester Thank you!

According to Travis CI output.
This class must be `open`. https://github.com/giginet/CustomKeyboardTextField/pull/2/commits/594287306ed19a2f4271868168b60f2824e19c57#diff-c99e96a90d698f7e9e026db431402453R15

I meant you should fix only properties, not this class.

I apologize confused you. Could you fix this?

**Comment by jollyjoester** [comment]
I'll continue to fix TravisCI error. Just a moment, please.

**Comment by giginet** [comment]
Thank you!!!!

CI is still fallen, however, I think it is caused because of building from the fork. 
Maybe, it will be fixed this branch is merged.

**Review by giginet — APPROVED** [review]

**Comment by jollyjoester** [comment]
Thank you very much!

**Pull request merged**